src/main/java/com/fabriccommunity/spookytime/world/feature/SpookyOreFeature.java

```diff
@@ -0,0 +1,137 @@
+
+import java.util.BitSet;
+import java.util.Random;
+import java.util.function.Function;
+
+public class SpookyOreFeature extends Feature<SpookyOreFeatureConfig> {
+    public SpookyOreFeature(Function<Dynamic<?>, ? extends SpookyOreFeatureConfig> function_1) {
+        super(function_1);
+    }
+
+    @Override
+    public boolean generate(IWorld world, ChunkGenerator<? extends ChunkGeneratorConfig> chunkGenerator, Random random, BlockPos blockPos, SpookyOreFeatureConfig oreFeatureConfig) {
+        float randomNumberFromZeroToPi = random.nextFloat() * 3.1415927F;
+        float dividedSize = (float)oreFeatureConfig.size / 8.0F;
+        int ceilSize = MathHelper.ceil(((float)oreFeatureConfig.size / 16.0F * 2.0F + 1.0F) / 2.0F);
+        double positiveX = (blockPos.getX() + MathHelper.sin(randomNumberFromZeroToPi) * dividedSize);
+        double negativeX = (blockPos.getX() - MathHelper.sin(randomNumberFromZeroToPi) * dividedSize);
+        double positiveZ = (blockPos.getZ() + MathHelper.cos(randomNumberFromZeroToPi) * dividedSize);
+        double negativeZ = (blockPos.getZ() - MathHelper.cos(randomNumberFromZeroToPi) * dividedSize);
+        double double_5 = (blockPos.getY() + random.nextInt(3) - 2);
+        double double_6 = (blockPos.getY() + random.nextInt(3) - 2);
+        int startX = blockPos.getX() - MathHelper.ceil(dividedSize) - ceilSize;
+        int y = blockPos.getY() - 2 - ceilSize;
+        int startZ = blockPos.getZ() - MathHelper.ceil(dividedSize) - ceilSize;
+        int xSize = 2 * (MathHelper.ceil(dividedSize) + ceilSize);
+        int int_7 = 2 * (2 + ceilSize);
+
+        for(int x = startX; x <= startX + xSize; ++x) {
+            for(int z = startZ; z <= startZ + xSize; ++z) {
+                if (y <= world.getTop(Type.OCEAN_FLOOR_WG, x, z)) {
+                    return this.generateVeinPart(world, random, oreFeatureConfig, positiveX, negativeX, positiveZ, negativeZ, double_5, double_6, startX, y, startZ, xSize, int_7);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    protected boolean generateVeinPart(IWorld world, Random random, SpookyOreFeatureConfig oreFeatureConfig, double positiveX, double negativeX, double positiveZ, double negativeZ, double double_5, double double_6, int startX, int yPosition, int startZ, int xSize, int int_5) {
+        int stonesPlaced = 0;
+        BitSet bitSet = new BitSet(xSize * int_5 * xSize);
+        Mutable blockPos = new Mutable();
+        double[] doubles_1 = new double[oreFeatureConfig.size * 4];
+
+        int counter;
+        double double_12;
+        double double_13;
+        double double_14;
+        double double_15;
+        for(counter = 0; counter < oreFeatureConfig.size; ++counter) {
+            float float_1 = (float)counter / (float)oreFeatureConfig.size;
+            double_12 = MathHelper.lerp((double)float_1, positiveX, negativeX);
+            double_13 = MathHelper.lerp((double)float_1, double_5, double_6);
```

Comment:
I assume by analogy that double_5 and double_6 are probably positiveY and negativeY, respectively. That would make double_12 something like currentX, double_13 currentY, and double_14 currentZ

Reply:
Also float_1 is probably `progress`